fix(agw): Handle WaitSetParameterValue correctly

amarpad opened the pull request:
## Summary
ACS clients can return non zero values for set calls on success (per std).
Make an optional flag to allow for SetParams to return non zero values.
Validated against FreedomFiOne and Baicells which return 1, 0 for
SetParameterValueResponse message.Status respectively.
Make change backwards compatible by using optional flag.


Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>

<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->



<!-- Enumerate changes you made and why you made them -->

## Test Plan

make test_python
Sercomm integration

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

Commit:
fix(agw): Handle WaitSetParameterValue correctly

ACS clients can return non zero values for set calls on success (per std).
Make an optional flag to allow for SetParams to return non zero values.
Validated against FreedomFiOne and Baicells which return 1, 0 for
SetParameterValueResponse message.Status respectively.
Make change backwards compatible by using optional flag.


Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>

ardzoht reviewed (approved):
lgtm

ardzoht commented on lte/gateway/python/magma/enodebd/state_machines/enb_acs_states.py:
```diff
@@ -918,19 +917,26 @@ def __init__(
-                    'Received SetParameterValuesResponse with '
-                    'Status=%d' % message.Status,
-                )
+            if not self.status_non_zero_allowed:
```
it may be worth to add a unit test for this case?

Commit:
fix(agw) Add a unit tests for WaitSetParameterValuesState

Add a unit test
Also creates a couple of dummy classes in case we want to extend the
test coverage to other state classes as well

Signed-off-by: Amar Padmanabhan <amar@freedomfi.com>
